### DISCUSSION TITLE
Advanced default parameter why task1 and task2 are the same?

### DISCUSSION THREAD

**Discussion by Harry040** [EA31337/EA31337#327] [Q&A] [answered]
EA_Task1_If=3; EA_Task1_Then=2; EA_Task2_If=3

https://github.com/EA31337/EA31337/releases/tag/v2.009

**Comment by kenorb** [comment, marked as answer]
It was surprising for me as well, but this was suggested by the optimization results at that time (especially for the lower deposit amounts).

Basically the task logic is: when equity reaches 5% above the balance, then close the most profitable order (a single one).
And the tasks are processed on each new minute (see: [ProcessTasks](https://github.com/EA31337/EA31337-classes/blob/v2.009/EA.mqh#L411)).

So when sudden price movement happens (especially within a minute), we can say that one task was not called frequent enough to handle the closure of overall profits, so the 2nd one is "helping" with closing profitable orders fast enough. Basically they're equivalent to action like: close two the most profitable orders. Secondly, when the profitable order is closed, in the next minute, the condition no longer apply, as you're no longer reaching 5%, so you require extra price movements before another close.

For example, your balance is $200, and after opening the market in the new week, your equity becomes $210 (4 orders having profit such as $1+$2+$3+$4 reaching 5% threshold). Having 1 task, you close 1 order with $4 profit, whereas having 2 same tasks, you close 2 orders with profit $7 (3+4) within the same minute. Now your account is no longer reaching 5% threshold, so you need to wait for the further price movement towards your profit to close the rest of the orders (and similar logic applies for the next time). However if the price goes opposite, 2 tasks basically did the better job.

**Reply by Harry040** [reply]
Thanks for you detailed reply!
